tests/integration/test_charm_nrpe.py

```diff
@@ -0,0 +1,66 @@
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Tests for jenkins relation with nrpe."""
+
+import pytest
+import pytest_asyncio
+from pytest_operator.plugin import OpsTest
+from ops.model import Application, ActiveStatus
+
+pytestmark = pytest.mark.nrpe
+
+
+NRPE_CHECK_HTTP_FILE = "/usr/lib/nagios/plugins/check_http"
+NRPE_CONFIG_FILE = "/etc/nagios/nrpe.d/check_check_jenkins_http.cfg"
+NAGIOS_CMD = "sudo $(grep -e '{}' {} | cut -d'=' -f2-)".format(
+    NRPE_CHECK_HTTP_FILE, NRPE_CONFIG_FILE
+)
+NAGIOS_CMD_SUCCESS = "HTTP OK"
+
+
+@pytest_asyncio.fixture(scope="module")
+async def nrpe(app_name: str, ops_test: OpsTest, app: Application, series: str):
+    """Add relationship with nrpe and nagios to app."""
+    nrpe_app: Application = await ops_test.model.deploy("nrpe", series=series)
+    await ops_test.model.add_relation(
+        "{}:nrpe-external-master".format(app_name), "nrpe:nrpe-external-master"
+    )
+    # [2022-09-29] Nagios does not support focal
+    await ops_test.model.deploy("nagios", series=(series if series != "focal" else "bionic"))
+    await ops_test.model.add_relation("nrpe:monitors", "nagios:monitors")
```

Comment:
You're adding a relation to nrpe using the nrpe-external-master relation, which means nrpe would be manually configured to connect to a nagios server (this relation was created before cross-model relations where available!). You're then also deploying nagios and relating nrpe to nagios. I think for what you're doing here (which is just executing nrpe checks on the jenkins unit itself) you don't need nagios at all. You can just deploy nrpe with the nrpe-external-master relation and that should be enough to configure the checks on the jenkins instance and you can then run the tests you're running.

Reply:
Will check this, from memory nrpe requires a nagios relation before it will start working, I'll check and add an explanatory comment

Reply:
Confirmed, nrpe requires nagios